### PR TITLE
ELPP-2427 Updated the notifications system to deal with taxonomy terms.

### PR DESCRIPTION
**Pull request by jamiehollern**
@nlisgo @giorgiosironi

### CODE REVIEW

File: src/modules/jcms_notifications/jcms_notifications.install [hunk 1]
```diff
@@ -1,5 +1,8 @@
 <?php
 
+use Drupal\jcms_notifications\MysqlNotificationStorage;
+use Drupal\Core\Database\Database;
+
 function jcms_notifications_schema() {
   $schema['jcms_notifications'] = [
     'description' => 'The base table for nodes.',
```
Comment (nlisgo) [inline src/modules/jcms_notifications/jcms_notifications.install]: For completeness this and other mentions of node should be changed to entity.

File: src/modules/jcms_notifications/jcms_notifications.install [hunk 2]
```diff
@@ -10,17 +13,34 @@ function jcms_notifications_schema() {
         'unsigned' => TRUE,
         'not null' => TRUE,
       ],
-      'node_id' => [
+      MysqlNotificationStorage::ID_FIELD => [
         'description' => 'The primary identifier for a node.',
```
Comment (nlisgo) [inline src/modules/jcms_notifications/jcms_notifications.install]: "a node" should be "an entity"

File: src/modules/jcms_notifications/jcms_notifications.install [hunk 3]
```diff
@@ -10,17 +13,34 @@ function jcms_notifications_schema() {
...
   return $schema;
 }
+
+/**
+ * Implements hook_update_N().
+ *
+ * Updates the node_id field to entity_id.
+ */
+function jcms_notifications_update_8101() {
+  $spec = [
```
Comment (nlisgo) [inline src/modules/jcms_notifications/jcms_notifications.install]: Could be:

```
$schema = jcms_notifications_schema();
$spec = $schema['jcms_notifications'][MysqlNotificationStorage::ID_FIELD];
```

Or just change "a node" to "an entity".

File: src/modules/jcms_notifications/src/NotificationStorageInterface.php [hunk 4]
```diff
@@ -18,31 +18,31 @@
...
-   * @param int $nodeId
+   * @param int $entityId
    *
    * @return null
    */
-  public function deleteNotificationNid(int $nodeId);
+  public function deleteNotificationId(int $entityId);
 
   /**
    * Takes an array of node IDs and deletes them.
```
Comment (nlisgo) [inline src/modules/jcms_notifications/src/NotificationStorageInterface.php]: should be "entity IDs"

File: src/modules/jcms_notifications/src/NotificationStorageInterface.php [hunk 5]
```diff
@@ -18,31 +18,31 @@
    *
    * @return int|null
    */
-  public function saveNotificationNid(EntityInterface $entity);
+  public function saveNotificationId(EntityInterface $entity);
 
   /**
    * Gets the node IDs from the notifications table then deletes them.
```
Comment (nlisgo) [inline src/modules/jcms_notifications/src/NotificationStorageInterface.php]: should be "entity IDs"

File: src/modules/jcms_notifications/src/NotificationStorageInterface.php [hunk 6]
```diff
@@ -18,31 +18,31 @@
...
    *
    * @return array
    */
-  public function getNotificationNids(): array;
+  public function getNotificationIds(): array;
 
   /**
    * Deletes a notification node ID.
```
Comment (nlisgo) [inline src/modules/jcms_notifications/src/NotificationStorageInterface.php]: should be "entity ID"

**Comment by jamiehollern** [comment]
@nlisgo Fixed the points you made.

File: src/modules/jcms_notifications/src/NotificationStorageInterface.php [hunk 7]
```diff
@@ -18,31 +18,31 @@
    *
    * @return int|null
    */
-  public function saveNotificationNid(EntityInterface $entity);
+  public function saveNotificationEntityId(EntityInterface $entity);
 
   /**
    * Gets the node IDs from the notifications table then deletes them.
```
Comment (nlisgo) [inline src/modules/jcms_notifications/src/NotificationStorageInterface.php]: Not sure how I missed these last time! Remove node from the method annotations in this file.

**Pull request merged**